src/api_manager/check_security_rules_test.cc

```diff
@@ -373,6 +399,137 @@ class CheckSecurityRulesTest : public ::testing::Test {
   std::string ruleset_test_url_;
 };
 
+// Cache hit without expiration. In this case, a cached "yes" is returned.
+TEST_F(CheckSecurityRulesTest, CheckAuthzCacheHitPositive) {
+  std::string service_config = std::string(kServiceName) + kProducerProjectId +
+                               kApis + kAuthentication + kHttp;
+  std::string server_config = kServerConfig;
+  SetUp(service_config, server_config);
+
+  request_context_->SetAuthToken(std::string(kAuthToken));
+  service_context_->authz_cache().Add(kCacheKey, true, kNow);
```

Comment:
it is better just test the external interface.  authz_cache() is internal implementation.  you should not access it.    

You should just call CheckSeurityRules() twice,  first one with cache miss, and second one with cache hit.